### PR TITLE
rsyslog/Sanity/gnutls-openssl: added anon mode and new/old syntax

### PR DESCRIPTION
**Pull request by sopos**
added variants for anon mode on client, server, and both sides
and all in old/new syntax variant

### CODE REVIEW

File: rsyslog/Sanity/gnutls-openssl/main.fmf [hunk 1]
```diff
@@ -2,55 +2,239 @@ summary: a sanity test for TLS communication
...
+#            extra-summary: rsyslogSyntax=new CLIENT_NO_CERT=1 /rsyslog/Sanity/gnutls-openssl
+#        /server_anon:
+#            environment+:
+#              SERVER_NO_CERT: 1
+#            extra-summary: rsyslogSyntax=new SERVER_NO_CERT=1 /rsyslog/Sanity/gnutls-openssl
+#        /both_anon:
+#            environment+:
+#              CLIENT_NO_CERT: 1
+#              SERVER_NO_CERT: 1
+#            extra-summary: rsyslogSyntax=new SERVER_NO_CERT=1 CLIENT_NO_CERT=1 /rsyslog/Sanity/gnutls-openssl
```
Comment (sarroutbi) [inline rsyslog/Sanity/gnutls-openssl/main.fmf:56]: Is all this code pretended to be left commented for some reason?

Comment (sopos) [reply rsyslog/Sanity/gnutls-openssl/main.fmf]: yes it was, I used that just for testing purposes, it is actually all covered by the individual virtual test cases

**Comment by sopos** [comment]
@sarroutbi do you agree with the rest of the changes?

**Comment by sarroutbi** [comment]
> @sarroutbi do you agree with the rest of the changes?

Yes. If commented code is kept intentionally, changes LGTM

**Comment by sopos** [comment]
it is actually removed now

**Comment by Cropi** [comment]
[General idea] Maybe one thing we should consider is to change ```/etc/rsyslogd.d/``` to ```/etc/rsyslog.d/``` (omit the "d" char), as this is the default directory for rsyslog config files.

**Comment by sopos** [comment]
> [General idea] Maybe one thing we should consider is to change `/etc/rsyslogd.d/` to `/etc/rsyslog.d/` (omit the "d" char), as this is the default directory for rsyslog config files.

I agree, that should not cause any problems as the include say `*.conf` so it should not collide. We could also move the certificates to even better place - under /etc/pki/ which is the official system-wide store. However, there's more tests doing that so I propose to do the change at once for all the tests via separate PR(s).

**Comment by sopos** [comment]
@sarroutbi, @Cropi Btw, have you noticed that on Rawhide there are more _anon_ tests fail there while on Centos just those _server_anon_?

**Comment by Cropi** [comment]
> @sarroutbi, @Cropi Btw, have you noticed that on Rawhide there are more _anon_ tests fail there while on Centos just those _server_anon_?

For example looking at: https://artifacts.dev.testing-farm.io/a33fa5b7-8968-4833-bda8-7b7847142c4b/ - /rsyslog/Sanity/gnutls-openssl/ossl/new-syntax/x509_name
Patching the config file fails, which causes:
```
:: [ 11:10:59 ] :: [   FAIL   ] ::   Command 'rsyslogPrepareConf' (Expected 0, got 1)
:: [ 11:11:00 ] :: [   FAIL   ] ::   Not found entered pattern GLOBALS 
:: [ 11:11:00 ] :: [   FAIL   ] ::   Not found entered pattern MODULES 
:: [ 11:11:00 ] :: [   FAIL   ] :: config client 
```

**Comment by sopos** [comment]
@Cropi I though I changed it to replace rather to patch everywhere, apparently not.. I'll take a look.

**Comment by sopos** [comment]
well, the patch was in _master_ already, rebased

**Comment by sopos** [comment]
@Cropi it seems there was a change in the rsyslog in the recent fedora builds - the timestamp template is not specified in the config

https://bugzilla.redhat.com/show_bug.cgi?id=2078621

**Pull request merged**